.github/workflows/pr-checks.yml

```diff
@@ -0,0 +1,34 @@
+name: Lint & Test
+
+on:
+  pull_request:
+    # edited is needed because that's the trigger when the base branch is
+    # changed on a PR
+    # The rest are the defaults.
+    types: [edited, opened, synchronize, reopened]
+
+jobs:
+  lint_and_test:
+    name: Lint & Test
+    env:
+      CI: true
```

Comment:
Doesn't Github set this automatically? https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables